src/figuresClasses.ts

```diff
@@ -1,19 +1,103 @@
+enum Colors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
 export interface Figure {
+  shape: Shape;
+  color: Colors;
+  getArea(): number;
+}
+
+function isNotValidTriangle(...shapeSide: number[]): boolean {
+  const sumSides = shapeSide
+    .reduce((sideOne: number, sideTwo: number) => sideOne + sideTwo);
 
+  return shapeSide.some((side: number) => side >= (sumSides - side));
 }
 
-export class Triangle {
+function isValidLength(...lengths : number[]): boolean {
+  return Math.min(...lengths) <= 0;
+}
 
+function roundingArea(area: number): number {
+  return Math.floor(area * 100) / 100;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  readonly shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (isValidLength(a, b, c)) {
+      throw new Error('Length of triangle sides should be greater than 0');
+    }
 
+    if (isNotValidTriangle(a, b, c)) {
+      throw new Error('The longest side of the triangle must be shorter than'
+       + ' the sum of the other two');
+    }
```

Comment:
u can move it to some private method

Reply:
Why? We use it only in triangle

Reply:
and I propose to make it a private method of the Triangle class 🙂